jwst/associations/lib/rules_level2b.py

```diff
@@ -69,11 +69,20 @@ def __init__(self, *args, **kwargs):
             Constraint_Base(),
             Constraint_Mode(),
             Constraint_Image_Science(),
-            Constraint_Single_Science(self.has_science),
             Constraint(
                 [Constraint_TSO()],
                 reduce=Constraint.notany
-            )
+            ),
```

Comment:
I've looked into this further, and it is a bug in `Constraint_Single_Science`. However, fixing that would require a bit of refactoring of a couple of rules.

Since this mimics what is happening for spectra, I say let's go with this. An issue will be filed for the constraint bug.